skafka/src/main/scala/com/evolutiongaming/skafka/consumer/ConsumerMetrics.scala

```diff
@@ -115,50 +117,226 @@ object ConsumerMetrics {
+      labels = LabelNames("client", "topic", "type")
+    )
+
+    val resultCounter = registry.counter(
+      name   = s"${prefix}_results",
+      help   = "Topic call result: success or failure",
+      labels = LabelNames("client", "topic", "type", "result")
+    )
+
+    val latencyHistogram = registry.histogram(
+      name    = s"${prefix}_latency",
+      help    = "Topic call latency in seconds",
+      buckets = latencyBuckets,
+      labels  = LabelNames("client", "topic", "type")
+    )
+
+    val recordsHistogram = registry.histogram(
+      name    = s"${prefix}_poll_records",
+      help    = "Number of records per poll",
+      buckets = pollCountBuckets,
+      labels  = LabelNames("client", "topic")
+    )
+
+    val bytesHistogram = registry.histogram(
+      name    = s"${prefix}_poll_bytes",
+      help    = "Number of bytes per poll",
+      buckets = pollBytesBuckets,
+      labels  = LabelNames("client", "topic")
+    )
+
+    val rebalancesCounter = registry.counter(
+      name   = s"${prefix}_rebalances",
+      help   = "Number of rebalances",
+      labels = LabelNames("client", "topic", "type")
+    )
+
+    val topicsLatency = registry.histogram(
+      name    = s"${prefix}_topics_latency",
+      help    = "List topics latency in seconds",
+      buckets = latencyBuckets,
+      labels  = LabelNames("client")
+    )
+    val ageHistogram = registry.histogram(
+      name    = s"${prefix}_poll_age",
+      help    = "Poll records age, time since record.timestamp",
+      buckets = pollAgeBuckets,
+      labels  = LabelNames("client", "topic")
+    )
+
+    for {
+      callsCounter      <- callsCounter
+      resultCounter     <- resultCounter
+      latencyHistogram  <- latencyHistogram
+      recordsHistogram  <- recordsHistogram
+      bytesHistogram    <- bytesHistogram
+      rebalancesCounter <- rebalancesCounter
+      topicsLatency     <- topicsLatency
+      ageHistogram      <- ageHistogram
+    } yield { (clientId: ClientId) =>
+      new Histograms(
+        callsCounter,
+        resultCounter,
+        latencyHistogram,
+        recordsHistogram,
+        bytesHistogram,
+        rebalancesCounter,
+        topicsLatency,
+        ageHistogram,
+        clientId
+      )
+    }
+  }
+  private val latencyBuckets =
+    Buckets(NonEmptyList.of(250e-6, 500e-6, 2e-3, 5e-3, 20e-3, 50e-3, 200e-3, 500e-3, 1, 2, 5, 30, 60))
+  private val pollCountBuckets =
+    Buckets(NonEmptyList.of(1, 20, 50, 200, 500))
+  private val pollBytesBuckets =
+    Buckets(
+      NonEmptyList.of(
+        2 * 1024,
+        32 * 1024,
+        64 * 1024,
+        128 * 1024,
+        512 * 1024,
+        1024 * 1024, // 1 MB – max record size
+        2 * 1024 * 1024,
+        32 * 1024 * 1024,
+        512 * 1024 * 1024, // 500 records per poll – the max poll size defaults
+      )
+    )
+  private val pollAgeBuckets =
+    Buckets(NonEmptyList.of(10e-3, 20e-3, 50e-3, 200e-3, 500e-3, 2, 5, 20, 60, 120))
```

Comment:
what are the units? reading `250e-6` as `250 nanoseconds` is not natural for most developers, please consider to provide at least comments like done for `1MB` or use something different, like `250.nanoseconds.toUnit(TimeUnit.SECONDS)`

Reply:
Added docs. I based these buckets on what we observe currently with summaries and our timeouts. We can also drop microseconds here, I personally don't see much value in them, but some might be worried by loss of precision at lower tail

Reply:
`pollAgeBuckets` are also in seconds? do those values have any meaning, like why there is `120 seconds` (2 minutes)?

Reply:
Everything regarding time is measured in seconds as I figured from code. I'd like for it to stay that way.

poll age is the age of records received from a poll call. 2 minutes is already pretty old for a record and it's good enough to signal that there is (was) a problem. It could happen after a downtime or after a particularly long rebalance.